From 05c673f588aee73a1d052353797fb960418e84d5 Mon Sep 17 00:00:00 2001
From: Anil Dsouza <anil.dsouza@einfochips.com>
Date: Sun, 6 Mar 2022 23:58:18 +0530
Subject: [PATCH 10/11] media: i2c : ap1302: Added support for multiple media
 bus formats

Added support for following media bus formats
1. MEDIA_BUS_FMT_UYVY8_2X8
2. MEDIA_BUS_FMT_RGB565_2X8_LE
3. MEDIA_BUS_FMT_JPEG_1X8

Change-Id: Id72b95f8970e59107371759685a04376a3093120
Signed-off-by: Anil Dsouza <anil.dsouza@einfochips.com>
---
 drivers/media/i2c/ap1302.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/media/i2c/ap1302.c b/drivers/media/i2c/ap1302.c
index 72a275897..200d8b762 100644
--- a/drivers/media/i2c/ap1302.c
+++ b/drivers/media/i2c/ap1302.c
@@ -481,11 +481,21 @@ struct ap1302_firmware_header {
 #define MAX_FW_LOAD_RETRIES 3
 
 static const struct ap1302_format_info supported_video_formats[] = {
+	{
+		.code = MEDIA_BUS_FMT_UYVY8_2X8,
+		.out_fmt = AP1302_PREVIEW_OUT_FMT_FT_YUV_JFIF
+			 | AP1302_PREVIEW_OUT_FMT_FST_YUV_422,
+	},
 	{
 		.code = MEDIA_BUS_FMT_RGB565_2X8_LE,
 		.out_fmt = AP1302_PREVIEW_OUT_FMT_FT_RGB
 			 | AP1302_PREVIEW_OUT_FMT_FST_RGB_565,
 	},
+	{
+		.code = MEDIA_BUS_FMT_JPEG_1X8,
+		.out_fmt = AP1302_PREVIEW_OUT_FMT_FT_JPEG_422
+			 | AP1302_PREVIEW_OUT_FMT_FST_JPEG_EXIF,
+	},
 };
 
 /* -----------------------------------------------------------------------------
-- 
2.17.1

